api/evidence/evidence.go

```diff
@@ -41,43 +42,20 @@ func (req *StoreEvidenceRequest) GetPayload() proto.Message {
 
 func (ev *Evidence) GetOntologyResource() ontology.IsResource {
 	var (
-		m        proto.Message
 		resource ontology.IsResource
-		err      error
 		ok       bool
 	)
-	// TODO: find a smarter way, because now we are unmarshalling the resource twice
-	// Try to extract the resource out of the evidence
-	m, err = ev.Resource.UnmarshalNew()
-	if err != nil {
+
+	if ev.Resource == nil || ev.Resource.Type == nil {
```

Comment:
Would it make sense to use `util.IsNil` here as well?

Reply:
No, since ev.Resource is not an interface and we only need the utility function to check for nil on interfaces.